utils.go

```diff
@@ -103,34 +102,37 @@ func RemoveDot(input string) string {
 }
 
 // LoadAndListenConfig loads config file to struct and listen changes in it.
-func LoadAndListenConfig(path string, obj interface{}, onUpdate func(oldObj interface{})) error {
+func LoadAndListenConfig[Conf any](path string, c Conf, onUpdate func(c Conf)) (Conf, error) {
```

Comment:
Documentation here about the safe usage of this function, and maybe a small example of mutex usage